pkg/webhook/server/webhook_cert_init.go

```diff
@@ -60,13 +65,19 @@ func syncService(ctx context.Context, crclient client.Client, namespace string)
 		if !apierrors.IsAlreadyExists(err) {
 			return err
 		}
-		existingCfg := &v1.Service{}
+		existingCfg := &corev1.Service{}
 		err := crclient.Get(ctx, types.NamespacedName{
 			Name:      secureService.Name,
 			Namespace: secureService.Namespace,
 		}, existingCfg)
+		if err != nil {
+			return err
+		}
+		clusterIP := existingCfg.Spec.ClusterIP
+		existingCfg.Spec = secureService.Spec
+		existingCfg.Spec.ClusterIP = clusterIP
 
-		err = crclient.Update(ctx, secureService)
+		err = crclient.Update(ctx, existingCfg)
```

Comment:
It means that we may lose critical annotation `service.beta.openshift.io/serving-cert-secret-name`.
I suppose it might be a safer option to restore ClusterIP in new object
```go
		secureService.ResourceVersion = existingCfg.ResourceVersion
		// ClusterIP is immutable and assigned by cluster
		secureService.Spec.ClusterIP = existingCfg.Spec.ClusterIP
		err = crclient.Update(ctx, secureService)
```

Reply:
I can copy in annotations and labels, the problem is that most of the metadata fields are populated by the cluster and we don't necessarily want to overwrite those (e.g. resourceVersion, etc.).

Reply:
Add step of copying labels/annotations over to existingCfg.

Reply:
I'm not sure about the use-case of keeping existing labels and annotation for che-workspace-controller service, like who is supposed to fill in them? Admin? Won't it break installation?
But this PR solves quite annoying bug and I'm OK to proceed with the proposed solution. We're able to change it later if we really need.